doc/html/pcre2api.html

```diff
@@ -2223,6 +2234,12 @@ <h1>pcre2api man page</h1>
 returning PCRE2_ERROR_CALLOUT_CALLER_DISABLED. This is useful if the application
 knows that a callout will not be provided to <b>pcre2_match()</b>, so that
 callouts in the pattern are not silently ignored.
+<pre>
+  PCRE2_EXTRA_TURKISH_CASING
+</pre>
+This option alters case-equivalence of the 'i' letters to follow the
+alphabet used by Turkish and Azeri languages. The option can be changed within
+a pattern by the (?t) option setting.
```

Comment:
Is this that important to reserve a letter for it? I don't remember anybody requested this before.

Reply:
I don't feel it's important. I just added it for consistency with `(?r)`. I'm very happy either way.

Reply:
to avoid the probability of conflict with a future Perl option, it was mentioned before that an uppercase letter would be preferred.

I am having a hard time seeing use cases for this though, so it would be wiser to hold this until those cases are presented IMHO.

Reply:
One obvious use case is someone writing a regex who does not have access to the code, but who wants to do the Turkish thing. They can start the pattern with (*UTF)(?t) but I guess this is pretty rare. But who knows?

Reply:
> They can start the pattern with (*UTF)(?t)

correct, but as you mentioned elsewhere for that use case it would be better if it is named `(*TURKISH_CASE)` or something similar which would have also the added benefit of not conflicting with Perl's future syntax so easily.

Reply:
I just want to mention that nobody complained about turkish casing so far. I would wait until somebody needs it.

Reply:
We are talking about reserving a letter. The feature is ok. I hope we can even find ways to optimize it to reduce its overhead.

Reply:
Sorry, I see, thank you!

If there isn't a letter reserved it should have `(*TURKISH)` and `(*NON_TURKISH)` as pre-pattern options.

I'll just await whatever decision you and Philip prefer. There's no "wrong" decision either way.

Reply:
I would not mind if some options could be anywhere in a pattern, not just at the beginning. We have full assertion support now. `(*TURKISH)` would work like `(?t)` and `(*TURKISH:pattern)` like `(?t:pattern)`.

Reply:
I have now forgotten the context, but some time ago there was a worry about the non-Perl things like (*UTF), but whoever it was that was worried was happy when they learned that these "specials" are recognized only at the start of a pattern as a sort of "pre-pattern". I would therefore confine (*TURKISH) and (*NON_TURKISH) in the same way. Perhaps that is best, just in case Perl invents (?t) at some point. I probably should have done (*CASELESS_RESTRICT) instead of (?r) but it is too late now.

Reply:
> I probably should have done (*CASELESS_RESTRICT) instead of (?r) but it is too late now.

Is it too late? It was added back in February, so I guess a fair number of users have already upgraded.

We could change it so that `(?r)` is recognised as a "pre-pattern" option only, with a big notice in the release notes, and add `(*CASELESS_RESTRICT)` as a new synonym. Then we can say `(?r)` is deprecated and in a future release it could be removed.

The reality is very few people, if any at all, are likely to be using it in the middle of a pattern, so this should be safe enough.

You've made breaking changes before, to align with Perl, but with flags to allow users to keep the old behaviour (eg `PCRE2_EXTRA_ALLOW_LOOKAROUND_BSK`). This would be similar - a small break but with simple workarounds for affected users.

It would very, very nice if both TURKISH_CASING and CASELESS_RESTRICT could be global options. Then all comparisons in the pattern would use the same case-folding rules. We wouldn't need the "OP_REFI" hack I added to propagate the casefolding flags; we wouldn't need OP_2CHAR (we'd have the option to just use ordinary OP_CHARI).

The more I think about it, the more I like the idea of having all casefolding be done consistently with one `UCD_CASEFOLD()` macro or similar, so there's no confusion.

Reply:
> > I probably should have done (*CASELESS_RESTRICT) instead of (?r) but it is too late now.
> 
> Is it too late?

Yes, and unlike TURKISH_CASING, it wouldn't work, because it is not a full flag, but meant to be used together with (?i). That might had been arguably another mistake, but there is no point arguing about it since it is now part of the API.

Luckily we can avoid making a mistake with TURKISH_CASING.

Reply:
I'm not convinced. Changing `(?r)` mid-pattern is so rare (and it's a new option this year, not an old one), we should at least consider removing that capability.

Reply:
On the contrary; it is very likely that it is mostly used "in pattern" because of the way it interacts with caseless. If I use it as a global it does nothing by itself, but if I use PCRE2_CASELESS globally then in every section of the pattern where I want to make sure only non Unicode characters match, then I can do `(?r:..)`. It even makes the matching faster because now all those sections are compiled to OP_CHARI.